extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/StaticResourcesBuildConfig.java

```diff
@@ -0,0 +1,20 @@
+package io.quarkus.vertx.http.runtime;
+
+import java.util.Map;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class StaticResourcesBuildConfig {
+
+    /**
+     * Whether to map directories from the local file system onto a specified path.
+     * <p>
+     * The key corresponds to the relative path of the directory on the file system, while the value corresponds
+     * to the HTTP path under which the resource should be served.
+     */
+    @ConfigItem
+    public Map<String, String> additionalLocalDirectories;
```

Comment:
I prefer the programmatic approach to this.

There are many concerns with such an approach that explain why it was not made that way.  First, you cannot configure anything (and adding all the configuration properties is going to lead to a testing matrix nightmare), just the path. Then, the order of the path is essential, a duplicated path or overlapping path will result in weird behavior. 

I still do not get why we need this instead of the documented programmatic approach. At the end, the user needs to configure it properly. 

Also, it's paths. They should be configurable at runtime, not build time.